boot/zephyr/serial_adapter.c

```diff
@@ -192,9 +192,9 @@ static int
 boot_uart_fifo_init(void)
 {
 #ifdef CONFIG_BOOT_SERIAL_UART
-	uart_dev = device_get_binding(CONFIG_RECOVERY_UART_DEV_NAME);
+	uart_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_console));
 #elif CONFIG_BOOT_SERIAL_CDC_ACM
-	uart_dev = device_get_binding(CONFIG_USB_CDC_ACM_DEVICE_NAME "_0");
+	uart_dev = DEVICE_DT_GET_ONE(zephyr_cdc_acm_uart);
```

Comment:
Two questions:
 1) shouldn't the `if (uart_dev)` that follows the line be universal for both the `zephyr_console` and `zephyr_cdc_acm_uart`?
 2) is that if even needed? wouldn't the `uart_dev` assignment fail, at compile time,  if `DEVICE_DT_GET_ONE` could not be resolved?

Reply:
`if (uart_dev)` should be `if (!device_is_ready(uart_dev))`, updated